cmd/notification.go

```diff
@@ -59,94 +59,126 @@ func (sys *NotificationSys) GetPeerRPCClient(addr xnet.Host) *PeerRPCClient {
 	return sys.peerRPCClientMap[addr]
 }
 
+// NotificationPeerErr returns error associated for a remote peer.
+type NotificationPeerErr struct {
+	Host xnet.Host // Remote host on which the rpc call was initiated
+	Err  error     // Error returned by the remote peer for an rpc call
+}
+
 // DeleteBucket - calls DeleteBucket RPC call on all peers.
-func (sys *NotificationSys) DeleteBucket(bucketName string) map[xnet.Host]error {
-	errors := make(map[xnet.Host]error)
+func (sys *NotificationSys) DeleteBucket(bucketName string) []NotificationPeerErr {
+	errs := make([]NotificationPeerErr, len(sys.peerRPCClientMap))
 	var wg sync.WaitGroup
+	idx := 1
```

Comment:
`idx` should start with zero here, otherwise we will have index out of range problem.